perf/Dockerfile

```diff
@@ -24,53 +21,62 @@ FROM debian:bullseye as intermediate
+    git clone --depth 1 --branch master https://github.com/openssl/openssl.git ossl-src && \
+    git clone --depth 1 --branch main https://github.com/open-quantum-safe/oqs-provider.git
+
+# Build OpenSSL3; set softlink to lib dir for cmake FindPackage
+WORKDIR /opt/ossl-src
+RUN LDFLAGS="-Wl,-rpath -Wl,${INSTALLDIR}/lib64" ./config shared ${OPENSSL_BUILD_DEFINES} --prefix=${INSTALLDIR} && \
+    make ${MAKE_DEFINES} && make install_sw && make install_ssldirs && ln -s ${INSTALLDIR}/lib64 ${INSTALLDIR}/lib;
 
 # build liboqs shared and static, distributable (x86_64)
 WORKDIR /opt/liboqs
 # base platform build (x86_64)
-RUN mkdir build && cd build && cmake .. ${LIBOQS_BUILD_DEFINES} -DOQS_DIST_BUILD=ON -DBUILD_SHARED_LIBS=ON -DCMAKE_INSTALL_PREFIX=/opt/ossl-src/oqs && make ${MAKE_DEFINES} && make install
-RUN mkdir build-static && cd build-static && cmake .. ${LIBOQS_BUILD_DEFINES} -DOQS_DIST_BUILD=ON -DBUILD_SHARED_LIBS=OFF -DCMAKE_INSTALL_PREFIX=/opt/ossl-src/oqs && make ${MAKE_DEFINES} && make install
+RUN mkdir build && cd build && cmake -GNinja .. ${LIBOQS_BUILD_DEFINES} -DOPENSSL_ROOT_DIR=${INSTALLDIR} -DOQS_DIST_BUILD=ON -DBUILD_SHARED_LIBS=ON -DCMAKE_INSTALL_PREFIX=${INSTALLDIR} && ninja && ninja install
+RUN mkdir build-static && cd build-static && cmake -GNinja -DOPENSSL_ROOT_DIR=${INSTALLDIR} .. ${LIBOQS_BUILD_DEFINES} -DOQS_DIST_BUILD=ON -DBUILD_SHARED_LIBS=OFF -DCMAKE_INSTALL_PREFIX=${INSTALLDIR} && ninja && ninja install
 
-# Build ref-only variants and store in INSTALLDIR-ref (x86_64)
-RUN mkdir build-shared-ref && cd build-shared-ref && cmake .. ${LIBOQS_BUILD_DEFINES} -DOQS_DIST_BUILD=OFF -DBUILD_SHARED_LIBS=ON -DOQS_OPT_TARGET="generic" -DCMAKE_INSTALL_PREFIX=${INSTALLDIR}/oqs-ref && make ${MAKE_DEFINES} && make install
-RUN mkdir build-static-ref && cd build-static-ref && cmake .. ${LIBOQS_BUILD_DEFINES} -DOQS_DIST_BUILD=OFF -DBUILD_SHARED_LIBS=OFF -DOQS_OPT_TARGET="generic" -DCMAKE_INSTALL_PREFIX=${INSTALLDIR}/oqs-ref && make ${MAKE_DEFINES} && make install
+# Build oqs-provider distributable
+WORKDIR /opt/oqs-provider
+RUN ln -s ../ossl-src openssl && cmake -DOPENSSL_ROOT_DIR=${INSTALLDIR} -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH=${INSTALLDIR} -S . -B _build && cmake --build _build  && cp _build/lib/oqsprovider.so ${INSTALLDIR}/lib64/ossl-modules/oqsprovider-plain.so && sed -i "s/default = default_sect/default = default_sect\noqsprovider = oqsprovider_sect/g" /opt/oqssa/ssl/openssl.cnf && sed -i "s/\[default_sect\]/\[default_sect\]\nactivate = 1\n\[oqsprovider_sect\]\nactivate = 1\n/g" /opt/oqssa/ssl/openssl.cnf && sed -i "s/providers = provider_sect/providers = provider_sect\nssl_conf = ssl_sect\n\n\[ssl_sect\]\nsystem_default = system_default_sect\n\n\[system_default_sect\]\nGroups = \$ENV\:\:DEFAULT_GROUPS\n/g" /opt/oqssa/ssl/openssl.cnf && sed -i "s/HOME\t\t\t= ./HOME           = .\nDEFAULT_GROUPS = kyber512/g" /opt/oqssa/ssl/openssl.cnf
```

Comment:
Would it be better to keep a copy of the desired `openssl.cnf` and replace the default one with it as opposed to this sed expression?

Reply:
Using a "disconnected" OQS-only config file would let us "miss" upstream changes to config options (and in such case can lead to long debugging sessions -- happened to me). Using `sed` in turn ensures we're always in sync with upstream master in this regard. And if an error running the expression occurs, we know precisely where to look and can understand what's new.

A fixed config file IMO would be preferable only if we also tie down the openssl code used, i.e., not use master. And _that_ would not be sensible given we need pretty current `openssl` code to run this.